vscode/src/prompt-builder/unique-context.ts

```diff
@@ -70,10 +70,18 @@ export function isUniqueContextItem(itemToAdd: ContextItem, uniqueItems: Context
             // Duplicates if overlapping ranges on the same lines,
             // or if one range contains the other.
             if (item.range && itemToAddRange) {
-                return !(
+                if (
                     rangesOnSameLines(item.range, itemToAddRange) ||
                     rangeContainsLines(item.range, itemToAddRange)
-                )
+                ) {
+                    return false
+                }
+            }
+
+            // Duplicates if whole file (undefined range) and selection has the
+            // same content.
+            if (item.range && !itemToAddRange && equalContent(item, itemToAdd)) {
```

Comment:
```suggestion
            if (!itemToAddRange && equalContent(item, itemToAdd)) {
```

This is my bad, just noticed we already exit if `!item.range` on line 61.

Reply:
Cool, will update this and line 72.